### PR TITLE
feat(Popup)!: migrate to floating-ui

### PR DESCRIPTION
**Pull request by oynikishin**

### CODE REVIEW

File: src/components/Popup/README.md [hunk 1]
```diff
@@ -8,13 +8,15 @@
 import {Popup} from '@gravity-ui/uikit';
 ```
 
-`Popup` can be used to display floating content above the page. It is a wrapper around [Popper.js](https://popper.js.org)
+`Popup` can be used to display floating content above the page. It is a wrapper around [floating-ui](https://floating-ui.com)
```
Comment (amje) [inline src/components/Popup/README.md:11]:
```suggestion
`Popup` can be used to display floating content above the page. It is a wrapper around [Floating UI](https://floating-ui.com)
```

File: src/components/Popup/README.md [hunk 2]
```diff
@@ -8,13 +8,15 @@
...
 with some defaults. To manage `Popup` visibility, use the `open` property.
 The `Popup` child components are rendered inside the [`Portal`](../Portal) component. To disable this behavior, use the `disablePortal` property.
 
 ## Anchor
 
-Ref object of the DOM element is passed to the `anchorRef` property to create a `Popper.js` instance.
+To specify the anchor of a floating element, you can use either the `anchorEl` or `anchorRef` property.
+The `anchorRef` property accepts a ref object of a DOM element.
```
Comment (amje) [inline src/components/Popup/README.md:18]: We will deprecate `anchorRef`, let's remove occurrences in the docs

File: src/components/Popup/types.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import type {OffsetOptions, Placement, VirtualElement} from '@floating-ui/react';
+
+import type {AUTO_PLACEMENTS} from './constants';
+
+export type PopupPlacement = (typeof AUTO_PLACEMENTS)[number] | Placement | Placement[];
+
+export type PopupAnchorEl = Element | VirtualElement;
```
Comment (amje) [inline src/components/Popup/types.ts:7]: Let's name this type fully: `PopupAnchorElement`

**Review by amje — APPROVED** [review]

**Review by Raubzeug — APPROVED** [review]

**Review by korvin89 — APPROVED** [review]

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Popup/Popup.tsx [hunk 4]
```diff
@@ -128,27 +198,49 @@ export function Popup({
...
-        placement,
-        // Take arrow size into offset account
-        offset: hasArrow ? [offset[0], offset[1] + ARROW_SIZE] : offset,
+    const {
+        refs,
+        floatingStyles,
+        placement: actualPlacement,
+        middlewareData,
+    } = useFloating({
+        floatingContext,
```
[the remark this answers is not in the document]
Comment (oynikishin) [reply src/components/Popup/Popup.tsx:212]: As far as I know, we plan to use floating-ui Interactions in high-level components that use Popup

[the remark this answers is not in the document]
Comment (amje) [reply src/components/Popup/Popup.tsx:212]: @ValeraS We will use `useInteractions` in Tooltip, Popover so we need this

**Comment by oynikishin** [comment]
Add changes to FilePreview because without them the tests would not run locally

**Review by ValeraS — APPROVED** [review]

**Pull request merged**